test/crc32_test.c

```diff
@@ -0,0 +1,202 @@
+/* crc32_test.c -- crc32 unit test
+ * Copyright (C) 2019-2021 IBM Corporation
+ * Authors: Rogerio Alves    <rogealve@br.ibm.com>
+ *          Matheus Castanho <msc@linux.ibm.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+typedef unsigned long crc32_type;
+#else
+#  include "zlib-ng.h"
+typedef uint32_t crc32_type;
+#endif
+
+typedef struct {
+    uint32_t line;
+    crc32_type crc;
+    const unsigned char *buf;
+    size_t len;
+    crc32_type expect;
+} crc32_test;
+
+void test_crc32(crc32_type crc, const unsigned char *buf, size_t len, uint32_t chk, uint32_t line) {
+    uint32_t res = PREFIX(crc32_z)((uint32_t)crc, buf, len);
+    if (res != chk) {
+        fprintf(stderr, "FAIL [%d]: crc32 returned 0x%08X expected 0x%08X\n",
+                line, res, chk);
```

Comment:
This line doesn't seem longer than line 29.

Reply:
Fixed.